docs/smart-contracts/language-features.md

```diff
@@ -0,0 +1,618 @@
+- A single semicolon (;) for short notes on a single line of code. 
+  
+  ```pact
+  ; First Pact module
+  (module greeting GOVERNANCE
+      (defcap GOVERNANCE () true)
+        (defun say-hello(name:string)
+          (format "Hello, {}! ~ from Kadena" [name])
+        )
+   )
+  
+  (say-hello "Pistolas") ; Call the hello function
+  ```
+
+- Two semicolons (;;) inside or above definitions that describe functions or other top-level forms.
+
+  ```pact
+  ;;  In this example, the module defines a table for storing greeting 
+  ;;  names and two functions:
+  ;; 
+  ;;  - (say-hello-to "name")
+  ;;  - (greet)
+  ```
+
+- Three semicolons (;;;) or more file-level comments or to separate larger sections of code.
+  
+  ```pact
+  ;;; Accounts module demonstrates using row-level keysets, private pacts, and escrow.
+  ;;; Version: 0.2 
+  ;;; Author: Stuart Popejoy
+  ```
+
+Depending on where you want to add a comment, you can also enclose strings using double quotation marks (`" "`) with or without the `@doc` metadata tag.
+The following is an example of a multi-line documentation string that describes the create-token function the :
+
+```pact
+@doc "Initializes a TOKEN with given policies and parameters, \
+\ and executes the enforce-init function for each listed policy."
+```
+ 
+The `@doc` metadata tag is optional.
+For example, a module definition, function definition, or table definition can include comments in strings without the @doc tag:
+
+```pact
+(module helloWorld 'admin-keyset
+  "A smart contract to greet the world."
+  (defun hello (name)
+    "Do the hello-world dance"
+    (format "Hello {}!" [name]))
+)
+...
+(deftable accounts:{account}
+    "Main table for accounts module.")
+```
+
+For more information about metadata, see [Pact syntax](/reference/syntax).
+
+## Data types
+
+Pact, like most programming languages, supports the data types that you would expect.
+For example, Pact allows you define the following types of data:
+
+| Data type | Description | Examples |
+| :--------- | :----------- | :------- |
+| Integer | Any whole number value—positive or negative—that doesn't include a decimal.| `1`, `2`, `-19` |
+| Decimal | Any number value that includes a decimal. Decimal values can have a potentially unlimited precision. | `1.0`, `23.5`, `3.14159265359` |
+| String | Any text within quotes. You can represent strings using double quotes. It's also possible to prepend strings used as function names or table names with a single quotation mark (').| `“Hello”`, `"Welcome to the show"`, `'balances` |
+| Boolean | Anything that is represented by true and false literals. | `true`, `false` |
+| List | List literals are created inside square brackets ([ ]). List items can be separated with spaces or commas. If all of the items in the list have the same type, then the type is defined by the content of the list. Otherwise, the type is just defined as a “list”. | `[1,2,3]` or `[1 2 3]`  is an integer list, `[1 2 true]` is a list |
```

Comment:
```suggestion
| List | List literals are created inside square brackets (`[ ]`). List items can be separated with spaces or commas. If all of the items in the list have the same type, then the type is defined by the content of the list. Otherwise, the type is just defined as a “list”. | `[1,2,3]` or `[1 2 3]`  is an integer list, `[1 2 true]` is a list |
```